src/api/app/views/webui/users/tokens/edit.html.haml

```diff
@@ -6,22 +6,57 @@
+              %fieldset.form-group
+                = f.label(:string_readonly, 'Your new OBS Personal Access Token Secret', class: 'col-form-label col-form-label-lg')
+                .input-group
+                  = render CopyToClipboardInputComponent.new(form_builder: f)
+              %hr
+
+          .form-row
             .col-12.col-md-10.col-lg-9
               .form-group
                 = f.label(:id, 'Id:')
                 = @token.id
               .form-group
                 = f.label(:type, 'Operation:')
                 = @token.class.token_name.capitalize
-              .form-group
-                = f.label(:scm_token) do
-                  SCM token:
-                  %abbr.text-danger{ title: 'required' } *
-                .input-group
-                  = f.password_field(:scm_token, size: 80, class: 'form-control', placeholder: 'Please enter your new SCM token', required: true)
+              - if @token.package.present?
+                .form-group
+                  = f.label(:type, 'Package:')
+                  = link_to("#{@token.package.project.name}/#{@token.package.name}",
+                    package_show_path(project: @token.package.project, package: @token.package))
+              .form-group#regenerate-token-group
+                = link_to('Regenerate Token', '#', title: 'Regenerate Token', class: 'btn btn-outline-danger float-right ml-3 px-4',
+                  data: { toggle: 'modal', target: '#regenerate-token-modal' })
+                If your token has been lost, forgotten or compromised, you can regenerate it.
+                Don't forget to replace the string in any scripts or applications using this token.
+              - if @token.type == 'Token::Workflow'
+                .form-group#scm-token-group
+                  = f.label(:scm_token) do
+                    SCM token:
+                    %abbr.text-danger{ title: 'required' } *
+                  .input-group
+                    = f.password_field(:scm_token, size: 80, class: 'form-control', placeholder: 'Please enter your new SCM token', required: true)
           .actions
             = link_to('Cancel', tokens_path, title: 'Cancel', class: 'btn btn-outline-secondary px-4 mr-3 mt-3 mt-sm-0')
-            = f.submit('Update', class: 'btn btn-primary px-4 mt-3 mt-sm-0')
+            - if @token.type == 'Token::Workflow'
+              = f.submit('Update', class: 'btn btn-primary px-4 mt-3 mt-sm-0')
+
+.modal.fade#regenerate-token-modal{ tabindex: -1, role: 'dialog', aria: { labelledby: 'regenerate-token-modal-label', hidden: true } }
+  .modal-dialog.modal-dialog-centered{ role: 'document' }
+    .modal-content
+      .modal-header
+        %h5.modal-title#regenerate-token-modal-label Do you really want to regenerate this token string?
+      .modal-body
+        %p
+          Please confirm that you want to regenerate the token string of the token with id
+          = surround "'" do
+            = @token.id
```

Comment:
The ID has no meaning to people, I would not mention it.

Reply:
Ok, removed.

Reply:
The ID is still here. Did you push the changes?

Reply:
Ops. I removed the wrong ID. Fixed.